src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -75,19 +76,18 @@ private static void runValidation(TransactionViewModel transactionViewModel, fin
         }
         for (int i = VALUE_TRINARY_OFFSET + VALUE_USABLE_TRINARY_SIZE; i < VALUE_TRINARY_OFFSET + VALUE_TRINARY_SIZE; i++) {
             if (transactionViewModel.trits()[i] != 0) {
-                //log.error("Transaction trytes: "+Converter.trytes(transactionViewModel.trits()));
                 throw new RuntimeException("Invalid transaction value");
             }
         }
 
         int weightMagnitude = transactionViewModel.weightMagnitude;
         if(weightMagnitude < minWeightMagnitude) {
-            /*
-            log.error("Hash found: {}", transactionViewModel.getHash());
-            log.error("Transaction trytes: "+Converter.trytes(transactionViewModel.trits()));
-            */
             throw new RuntimeException("Invalid transaction hash");
         }
+
+        if (transactionViewModel.value() != 0 && transactionViewModel.getAddressHash().trits()[Curl.HASH_LENGTH - 1] != 0) {
+            throw new RuntimeException("Invalid transaction address");
```

Comment:
Why if a address last trits isn't 0 will be invalid?

Reply:
yes,
this is already the case, due to Kerl conversion.
till now this was enforced in BundleValidation: https://github.com/iotaledger/iri/blob/dev/src/main/java/com/iota/iri/BundleValidator.java#L53

this is added to not propagate such transactions in the first place.